client/a8c-for-agencies/sections/sites/features/jetpack/jetpack-sites-dataviews.tsx

```diff
@@ -0,0 +1,401 @@
+import { Button, Gridicon } from '@automattic/components';
+import { Icon, starFilled } from '@wordpress/icons';
+import { useTranslate } from 'i18n-calypso';
+import { useCallback, useEffect, useContext, useMemo, useState, ReactNode } from 'react';
+import { GuidedTourStep } from 'calypso/a8c-for-agencies/components/guided-tour-step';
+import { DATAVIEWS_LIST } from 'calypso/a8c-for-agencies/components/items-dashboard/constants';
+import ItemsDataViews from 'calypso/a8c-for-agencies/components/items-dashboard/items-dataviews';
+import {
+	DataViewsColumn,
+	DataViewsState,
+	ItemsDataViewsType,
+} from 'calypso/a8c-for-agencies/components/items-dashboard/items-dataviews/interfaces';
+import SiteSort from 'calypso/a8c-for-agencies/components/items-dashboard/items-dataviews/site-sort';
+import SiteSetFavorite from 'calypso/a8c-for-agencies/sections/sites/site-set-favorite';
+import SitesDashboardContext from 'calypso/a8c-for-agencies/sections/sites/sites-dashboard-context';
+import {
+	SiteInfo,
+	SitesDataViewsProps,
+} from 'calypso/a8c-for-agencies/sections/sites/sites-dataviews/interfaces';
+import SiteDataField from 'calypso/a8c-for-agencies/sections/sites/sites-dataviews/site-data-field';
+import SiteActions from 'calypso/jetpack-cloud/sections/agency-dashboard/sites-overview/site-actions';
+import useFormattedSites from 'calypso/jetpack-cloud/sections/agency-dashboard/sites-overview/site-content/hooks/use-formatted-sites';
+import SiteStatusContent from 'calypso/jetpack-cloud/sections/agency-dashboard/sites-overview/site-status-content';
+import { JETPACK_MANAGE_ONBOARDING_TOURS_EXAMPLE_SITE } from 'calypso/jetpack-cloud/sections/onboarding-tours/constants';
+import TextPlaceholder from 'calypso/jetpack-cloud/sections/partner-portal/text-placeholder';
+import { AllowedTypes, Site, SiteData } from '../../types';
+
+export const JetpackSitesDataViews = ( {
+	data,
+	isLoading,
+	isLargeScreen,
+	setDataViewsState,
+	dataViewsState,
+	forceTourExampleSite = false,
+	className,
+}: SitesDataViewsProps ) => {
+	const translate = useTranslate();
+
+	const { showOnlyFavorites } = useContext( SitesDashboardContext );
+	const totalSites = showOnlyFavorites ? data?.totalFavorites || 0 : data?.total || 0;
+	const sitesPerPage = dataViewsState.perPage > 0 ? dataViewsState.perPage : 20;
+	const totalPages = Math.ceil( totalSites / sitesPerPage );
+
+	const sites = useFormattedSites( data?.sites ?? [] );
+
+	const openSitePreviewPane = useCallback(
+		( site: Site ) => {
+			setDataViewsState( ( prevState: DataViewsState ) => ( {
+				...prevState,
+				selectedItem: site,
+				type: DATAVIEWS_LIST,
+			} ) );
+		},
+		[ setDataViewsState, dataViewsState ]
+	);
```

Comment:
This is a common code that should be in an upper layer, not here. For now is okay.